### PR TITLE
View "to/from" messages for subscribers

### LINKED ISSUE #13: Admin can view all messages to/from a given number
Could go in the subscriber view, but it would be nice to be able to see a log of all messages between the system and a particular phone number.

### PR DESCRIPTION
**Pull request by imacrayon**
Fixes #13

### CODE REVIEW

File: app/Http/Controllers/SubscriberController.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    public function new()
+    {
+        return view('admin.subscribers.new', [
+            'subscriber' => new Subscriber(),
+        ]);
+    }
+
+    public function create(Request $request)
```
Comment (sethetter) [inline app/Http/Controllers/SubscriberController.php]: This scares me a bit. From a legal perspective, this service can't text people who haven't explicitly opted in. I'm probably being more cautious than is really necessary, but if we are going to allow new subscribers via the web UI, we might want to think about sending the subscribed message along with it so they can immediately opt out if they want. I'm still not sure if we'd be in full compliance with the law doing that though.

We can keep this around for now, but just maybe don't provide access to it through the UI until we know whether this is something we should avoid for sure.

**Comment by imacrayon** [comment]
@sethetter are you opposed to adding a `subscriber_id` column on the messages table? We can get all messages for a subscriber through the `to` and `from` columns, but doing it that way means we have two foreign keys, which complicates the code.

**Comment by imacrayon** [comment]
I've linked the numbers on the subscribers page to a "Messages" page that shows a filterable list of all  incoming/outgoing messages for the number.

I've added a database column and a `helpers.php` file to `tests` so you'll have to run `php artisan migrate` and `composer dump-autoload`.

**Comment by sethetter** [comment]
Question -- do you think it might be better to structure the relationship between `Message` and `Subscriber` using `number` somehow? This way if the subscriber record happens to get deleted, but recreated when they resubscribe, the old message history is still there?

**Comment by imacrayon** [comment]
Good point. I guess I hadn’t considered that because right now when a subscriber unsubscribes only the`subscribed` boolean is changed. Changing `subscriber_id` to `subscriber_number`seems like a sensible safety net though.

File: database/migrations/2018_07_05_183514_add_subscriber_number_to_messages.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('messages', function ($table) {
+            $column = $table->string('subscriber_number')->unsigned()->after('twilio_sid');
```
Comment (sethetter) [inline database/migrations/2018_07_05_183514_add_subscriber_number_to_messages.php]: Gotta drop the `->unsigned()` now

File: resources/views/admin/subscribers/form.blade.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
+@include ('partials.errors')
+
+<form id="subscriber_form" action="{{ $subscriber->exists ? route('subscribers.admin.update', $subscriber) : route('subscribers.admin.create') }}" method="POST">
+    @csrf
+    <div class="form-group">
+        <label class="form-label" for="body">Number</label>
+        <input type="text" class="form-control" name="number" id="number" value="{{ old('number', $subscriber->number) }}">
```
Comment (sethetter) [inline resources/views/admin/subscribers/form.blade.php]: Let's just display this instead of allowing it to be changed.

**Comment by sethetter** [comment]
Looks good! Awesome work @imacrayon, I learned several things from this PR :)

**Comment by imacrayon** [comment]
Glad I could help @sethetter and thanks for your feedback!

**Pull request merged**